### ISSUE TITLE
VA.gov Search Click Tracking API - Frontend Development

### ISSUE THREAD

**Issue by johnhashva** [department-of-veterans-affairs/va.gov-team#14955]
## Issue Description
Search.gov -- which powers VA.gov site search -- has completed development and soft launched API-based [click-tracking](https://search.gov/manual/clicks.html) to enable "closed loop" analytics.   They have asked VA.gov, along with other agencies, to test the capability.

The click tracking API endpoint lets you send in click events, which allows you to see click data on your Admin Click Analytics page. We also use this click data to strengthen the search results algorithm for your site.

The Click Track API instructions are attached for implementation and validation.

[Search.gov - Click Tracking API Instructions (Beta).docx](https://github.com/department-of-veterans-affairs/va.gov-team/files/5393167/Search.gov.-.Click.Tracking.API.Instructions.Beta.docx)

## Front end tasks
- Ensure the back end data from API is displayed in local and review instance
- Ensure unit tests are available. 
- Ensure your code changes are covered by E2E tests

## Acceptance Criteria

- [ ] Success - A response status code of 200 and empty body.

- [ ] Missing Required Parameters - A response status code of 400 and an error message describing the missing parameters. ["Query can't be blank"]

- [ ] Invalid Or Inactive Search Site Handle - A response status code of 401 and an error message. ["Affiliate is invalid"]

- [ ] Invalid API Access Key - A response status code of 401 and an error message. ["Access key is invalid"]

- [ ] Unparseable URL - A response status code of 401 and an error message. ["Url is not a valid format"]

* ●	Invalid Module Code - A response status code of 401 and an error message. ["Module code {MODULE} is not a valid module"]​

**Comment by ncksllvn** [comment]
So to implement this, we will have to edit our on-site Search app to send an API request to search.gov's newly built endpoint whenever a search result is clicked.

Per the docs - 

> The endpoint is `https://api.gsa.gov/technology/searchgov/v2/clicks/`
>
> A full example of this is `https://api.gsa.gov/technology/searchgov/v2/clicks?url={URL}&query={QUERY}&affiliate=logstash&position={POSITION}&module_code={MODULE_CODE}&access_key={ACCESS_KEY}=&client_ip={CLIENT_IP}&user_agent={USER_AGENT}`

We should verify that the docs are correct in stating the `access_key` is a required parameter, because I think that needs to be kept secret, so it would save us a decent amount of work if it weren't a required param.

**Comment by johnhashva** [comment]
@ncksllvn thanks for looking into this.  Do you have 15m to talk through this at some point this week? If yes, I'll find us that time via your calendar.

**Comment by johnhashva** [comment]
@ncksllvn  just heard back from Dawn on your question, her response:

Hi John, the access_key is definitely required, it's uniquely paired with the affiliate handle.

However, it's the same access_key as you use for your search calls, so perhaps for the click call the key can just be referenced from its current location? or managed in the same way?

Dawn

**Comment by denisecoveyduc** [comment]
To be assigned to Sam; can discuss with Nick Sullivan who has some ideas on how to implement . Week of Nov 2 - S&D team will have a meeting with search.gov to decide what comes next with click tracking. 

History: Search.gov has implemented click tracking and asked that we to some test validation.

**Comment by johnhashva** [comment]
Meeting is scheduled for Monday 11/2 at 4pm with Search.gov to determine (a) is there opportunity to test/validate the API and (b) to surface any questions (such as Nick's above) about how VA.gov implements this for testing.  Note: The Search.gov team is already implementing updates to click tracking based on test feedback and those changes should be ready by next week.

**Comment by denisecoveyduc** [comment]
As per sprint planning (Nov 5), @srsuddath will discuss this issue with @ncksllvn. What is involved in this issue and can we assign it for Sprint 34 or should we aim for Sprint 35?  Once knowledge transfer has occurred we can make an informed decision.

**Comment by denisecoveyduc** [comment]
@johnhashva will add update on discussion with team and close. Search & Discovery team will not meet the validation deadline. A new ticket will be created to pick up Click Tracking API  and connecting to production API.

**Comment by johnhashva** [comment]
@denisecoveyduc + @srsuddath an update from Search.gov.   They are willing to hold up production release until we have had an opportunity to test -- if we can get to it "right after" the Thanksgiving holiday.   Could you assess the LOE in the next week and get back her with a sense of our implementation timeline for validation testing?

**Comment by denisecoveyduc** [comment]
@johnhashva Just to confirm we are all on the same page. As per our discussion in slack on Wednesday @srsuddath plans to prioritize this issue over the work to finalize the type ahead POC [#14452 (likely to be completed)  and #13494 (a stretch goal)]. Let us know if that is not aligned to your expectations.

**Comment by denisecoveyduc** [comment]
Type Ahead POC Work was prioritized over the Click Tracking Validation in Sprint 35. As a result, this issue will move into Sprint 36.

**Comment by denisecoveyduc** [comment]
Accessibility issues for Type Ahead POC have once again taken priority as we are currently waiting on the Platform team to complete the back end development required to move forward with this issue. Once they have completed their portion (#17009) then @srsuddath can complete (#16367) once both are complete then this front end  will be addressed. Moving to Sprint 37.

**Comment by srsuddath** [comment]
I would like to further emphasize that this ticket cannot be validated until ticket (#17009) has been completed.

**Comment by srsuddath** [comment]
This ticket needs to slide to sprint 39, as the backend work for Click tracking has ballooned beyond our estimates. Until the backend work is done, this ticket will remain blocked.

**Comment by srsuddath** [comment]
Click tracking has been merged in!
